complex/cargo/list/src/items.rs

```diff
@@ -140,3 +145,44 @@ pub unsafe extern "C" fn item_label_at(label_list: *const Vec<Label>, index: siz
     let label = Box::new(label_list[index].clone());
     Box::into_raw(label)
 }
+
+#[cfg(target_os="android")]
+#[allow(non_snake_case)]
+pub mod android {
+    extern crate jni;
+
+    use super::*;
+    use self::jni::JNIEnv;
+    use self::jni::objects::{JClass, JString, JValue};
+    use self::jni::sys::{jlong};
+
+    #[no_mangle]
+    pub unsafe extern fn Java_com_mozilla_toodle_RustToodle_newItem(env: JNIEnv, _: JClass) -> jlong {
+        Box::into_raw(Box::new(new_item())) as jlong
+    }
+
+    #[no_mangle]
+    pub unsafe extern fn Java_com_mozilla_toodle_RustToodle_itemSetName(env: JNIEnv, _: JClass, item: *mut Item, name: JString) {
+        let item = &mut*item;
+        log("Got item");
+
+        // TODO line below crashes... why?
+        let new_name: String = env.get_string(name).expect("Couldn't get item name").into();
+        log("Got name");
+        log(&new_name);
+
+        item.name = new_name;
+    }
+
+    #[no_mangle]
+    pub unsafe extern fn Java_com_mozilla_toodle_RustToodle_itemSetDueDate(env: JNIEnv, _: JClass, item: *mut Item, due_date: JValue) {
+        let item = &mut*item;
```

Comment:
again, just call `item_set_due_date` rather than having to do it all again here.